firmware/midi_controller/Note.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef NOTE_HPP
+#define NOTE_HPP
+
+namespace GrandPiano
+{
+
+// MIDI note IDs
+#define MIDI_C4 48
+#define MIDI_D4 50
+#define MIDI_E4 52
+#define MIDI_F4 53
+#define MIDI_G4 55
+#define MIDI_A4 57
+#define MIDI_B4 59
+#define MIDI_C5 60
+#define MIDI_CSHARP4 49
+#define MIDI_DSHARP4 51
+#define MIDI_FSHARP4 54
+#define MIDI_GSHARP4 56
+#define MIDI_ASHARP4 58
+
+struct Note
```

Comment:
Personally, I would remove this data structure and split it up among other files. Velocity is a property of playback, not the note itself. Same with channel. They would really fit better being solely part of midi_controller.ino. ID makes more sense as a member of Key.

The MIDI_* defines are good though, and it might be worth keeping this file around just for them. Alternatively, if you think they would fit better in midi_controller.ino, feel free to move them there.